common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockStoreClient.java

```diff
@@ -256,6 +256,23 @@ public void onFailure(Throwable e) {
     }
   }
 
+  @Override
+  public boolean removeShuffleMerge(String host, int port, int shuffleId, int shuffleMergeId) {
+    checkInit();
+    try {
+      TransportClient client = clientFactory.createClient(host, port);
+      client.send(
+          new RemoveShuffleMerge(appId, comparableAppAttemptId, shuffleId, shuffleMergeId)
+              .toByteBuffer());
+      // TODO(SPARK-42025): Add some error logs for RemoveShuffleMerge RPC.
+    } catch (Exception e) {
+      logger.error("Exception while sending RemoveShuffleMerge request to {}:{}",
```

Comment:
Let us make it info in this pr - we can revisit it as part of SPARK-42025

Reply:
Do we need add an new RPC result type and add an callback here?

Reply:
We can revisit that in [SPARK-42025](https://issues.apache.org/jira/browse/SPARK-42025).
Here, let us simply make the `logger.error` in the `catch` to `logger.debug` for now - since it is this is a best case effort to remove files from an external service.